h/views/account_signup.py

```diff
@@ -76,15 +81,169 @@ def validation_failure(self):
 
     @property
     def js_config(self) -> dict[str, Any]:
-        return {"csrfToken": get_csrf_token(self.request)}
+        feature_service = self.request.find_service(name="feature")
+
+        return {
+            "csrfToken": get_csrf_token(self.request),
+            "features": {
+                "log_in_with_orcid": feature_service.enabled(
+                    "log_in_with_orcid", user=None
+                )
+            },
+        }
+
+
+class AuthJWTDecodeError(Exception):
+    """Decoding the `auth` JWT query param failed."""
+
+
+@dataclass
+class IDInfo:
+    """Information about the user's account with a third-party provider.
+
+    When signing up to Hypothesis with a third-party provider ("Sign up with
+    Google" etc) the IDInfo class represents the payload of a signed JWT that
+    we use to pass information about the user's account with the third-party
+    from one route of our app to another when redirecting the browser (we
+    include the JWT in a query param).
+
+    The standard JWT `sub` (subject) claim is used to contain the user's unique
+    ID from the third-party provider. The JWT spec says that `sub` must be
+    locally unique within the context of the issuer (the `iss` claim). We
+    achieve this by using a different `iss` for each provider (Google,
+    Facebook, etc).
+
+    https://www.rfc-editor.org/rfc/rfc7519#section-4.1.2
+
+    """
+
+    sub: str
```

Comment:
More fields will be added here in future. For example `email`, and later possibly some fields about the user's name that we might use to generate a suggested Hypothesis username.